### DISCUSSION TITLE
How to test a File-Picker?

### DISCUSSION THREAD

**Discussion by nadimhq** [go-rod/rod#419] [Q&A]
I want to be able to test File-Picker for UI testing.

Our application also uses drag and drop, it would be awesome if there was drag and drop feature. In our case if one of these work then the other will.

**Comment by ysmood** [comment]
Page.Mouse.Down, Page.Mouse.Move, Page.Mouse.Up doesn't work for you?

**Reply by nadimhq** [reply]
Can go-rod interact with the file browser of my operating system? So would drag and drop work outside of my browser?

**Reply by ysmood** [reply]
If you just want to upload some file:

https://github.com/go-rod/rod/blob/67764886ede23171b3892f69b38ce3c4c948243d/element_test.go#L515-L526
